lib/modules/datasource/metadata.ts

```diff
@@ -114,12 +127,16 @@ export function addMetaData(
   ) {
     dep.sourceUrl = dep.changelogUrl;
   }
-  // prettier-ignore
-  if (dep.homepage?.includes('github.com')) { // lgtm [js/incomplete-url-substring-sanitization]
-    if (!dep.sourceUrl) {
-      dep.sourceUrl = dep.homepage;
+
+  if (dep.homepage) {
+    const parsedHomePage = parseUrl(dep.homepage);
+    // prettier-ignore
+    if (parsedHomePage?.hostname.includes('github')) {  // lgtm [js/incomplete-url-substring-sanitization]
```

Comment:
comment still needed ?

Reply:
I'm not sure if they want that comment left untouched. but I don't think it is needed